aea/helpers/search/models.py

```diff
@@ -434,12 +681,30 @@ def check(self, description: Description) -> bool:
             value, type(next(iter(self.constraint_type.value)))
         ):
             return False
-        if not isinstance(value, type(self.constraint_type.value)):
+        if type(self.constraint_type.value) not in {
```

Comment:
this was missing @marcofavorito

Reply:
true. I think the pre-check on the type should be improved by delegating it to the constraint type class and handle each case.